utils/httputil/client.go

```diff
@@ -0,0 +1,24 @@
+package httputil
+
+import "net/http"
+
+// RetriableStatus returns true if the HTTP status code should be retried.
+// 4xx codes are considered non-retriable except for 408 - request timeout and 429 - too many requests.
+func RetriableStatus(statusCode int) bool {
+	// 2xx, 3xx
+	if statusCode < 400 {
+		return true
+	}
```

Comment:
why would anyone want to retry a 2xx or 3xx?

Reply:
Why not :P Ok, initially, this method has called NonRetriableStatus. By default, everything should be retried except 4xx (with two exceptions). Retries happen only after an error, so if your logic determines there is an error on a 2xx or 3xx you should be able to retry.

I understand that it creates confusion, so I will make them non-retriable as well.